libsolidity/formal/SMTChecker.cpp

```diff
@@ -614,6 +624,19 @@ void SMTChecker::resetVariables(vector<Declaration const*> _variables)
 	}
 }
 
+void SMTChecker::mergeVariables(vector<Declaration const*> _variables, smt::Expression _condition, VariableSequenceCounters& _countersEndTrue, VariableSequenceCounters& _countersEndFalse)
+{
+	for (auto const* decl: _variables)
+	{
+		newValue(*decl);
+		int trueCounter = _countersEndTrue.at(decl);
+		int falseCounter = _countersEndFalse.at(decl);
+		smt::Expression e = currentValue(*decl) == smt::Expression::ite(_condition, valueAtSequence(*decl, trueCounter), valueAtSequence(*decl, falseCounter));
```

Comment:
We talked about the fact that we have to take the full paths to the ends of the conditions into account, were you able to find out anything about that?

Reply:
We don't really need to use the full history of conditions in the `ite` vars, since that is implicitly done by the solver.
Example:
```
if (a == 0)
    b = 100;
else if (a == 1)
    b = 200;
else
    b = 300;
assert(b <= 400);
```
Here we'll have the definitions
```
- b_0 = 0
- b_1 = 100
- b_2 = 200
- b_3 = 300
- b_4 = ite(a == 1, b_2, b_3)
- b_5 = ite(a == 0, b_1, b_4)
```
The `assert`s SMT query contains the conjunction of the path conditions so far, and the solver will use those to compute the value of `b_5` (by computing the values of all `b_i`) and prove stuff.

Reply:
Ok, now I see! It is indeed better to just add `branch_cond -> require_cond` because we can then do what you do here and do not have to care about the reverting branches.

Reply:
Exactly, I mostly wanted to avoid caring about those branches.